File: browsable_source/Amaya-1-0b/kaffe-0.8.4/test/benchmarks/sieve/Sieve.java
//  Eratosthenes Sieve prime number benchmark in Java

public class Sieve {

  static String results1, results2;

  public static void main(String[] args) {
     System.out.println("Running Sieve benchmark.");
     System.out.println("This will take about 10 seconds.");
     runSieve();
     System.out.println(results1);
     System.out.println(results2);
  }

  static void runSieve() {
     int SIZE = 8190;
     boolean flags[] = new boolean[SIZE+1];
     int i, prime, k, iter, count;
     int iterations = 0;
     double seconds = 0.0;
     int score = 0;
     long startTime, elapsedTime;

     startTime = System.currentTimeMillis();
     while (true) {
        count=0;
        for(i=0; i<=SIZE; i++) flags[i]=true;
        for (i=0; i<=SIZE; i++) {
           if(flags[i]) {
              prime=i+i+3;
              for(k=i+prime; k<=SIZE; k+=prime)
                 flags[k]=false;
              count++;
           }
        }
        iterations++;
        elapsedTime = System.currentTimeMillis() - startTime;
        if (elapsedTime >= 10000) break;
     }
     seconds = elapsedTime / 1000.0;
     score = (int) Math.round(iterations / seconds);
     results1 = iterations + " iterations in " + seconds + " seconds";
     if (count != 1899)
        results2 = "Error: count <> 1899";
     else
        results2 = "Sieve score = " + score;
  }
}

File: browsable_source_code/Amaya-1-0b/kaffe-0.8.4/test/benchmarks/sieve/Sieve.c
static char results1[100];
static char results2[100];

static void runSieve(void);

main(int argc, char* argv)
{
     printf("Running Sieve benchmark.\n");
     printf("This will take about 10 seconds.\n");
     runSieve();
     printf(results1);
     printf(results2);
}

static
void
runSieve()
{
     int SIZE = 8190;
     char flags[SIZE+1];
     int i, prime, k, iter, count;
     int iterations = 0;
     double seconds = 0.0;
     int score = 0;
     long startTime, elapsedTime;

     time(&startTime);
     while (1) {
        count=0;
        for(i=0; i<=SIZE; i++) flags[i]=1;
        for (i=0; i<=SIZE; i++) {
           if(flags[i]) {
              prime=i+i+3;
              for(k=i+prime; k<=SIZE; k+=prime)
                 flags[k]=0;
              count++;
           }
        }
        iterations++;
	time(&elapsedTime);
        elapsedTime -= startTime;
        if (elapsedTime >= 10) break;
     }
     seconds = elapsedTime;
     score = (int) (iterations / seconds);
     sprintf(results1, "%d iterations in %f seconds\n", iterations, seconds);
     if (count != 1899)
        sprintf(results2, "Error: count <> 1899\n");
     else
        sprintf(results2, "Sieve score = %d\n", score);
}
